zim/plugins/linkmap.py

```diff
@@ -120,33 +131,41 @@ class LinkMapPageViewExtension(PageViewExtension):
 
 	def __init__(self, plugin, pageview):
 		PageViewExtension.__init__(self, plugin, pageview)
-		self.on_preferences_changed(plugin.preferences)
+		self.preferences = plugin.preferences
+		self.on_preferences_changed(self.preferences)
 		self.connectto(plugin.preferences, 'changed', self.on_preferences_changed)
 
 	def on_preferences_changed(self, preferences):
 		self.set_action_in_headerbar(self.show_linkmap, preferences['button_in_headerbar'])
 
 	@action(_('Link Map'), icon='linkmap-symbolic', menuhints='view:headerbar') # T: menu item
 	def show_linkmap(self):
-		linkmap = LinkMap(self.pageview.notebook, self.pageview.page)
-		dialog = LinkMapDialog(self.pageview, linkmap, self.navigation)
+		dialog = LinkMapDialog(self.pageview, self.navigation, self.preferences)
 		dialog.show_all()
 
 
 class LinkMapDialog(Dialog):
 
-	def __init__(self, parent, linkmap, navigation):
-		Dialog.__init__(self, parent, 'LinkMap',
+	def __init__(self, parent, navigation, preferences):
+		# LinkMap can't navigate across notebooks
+		self.title_ending = ' - ' + parent.notebook.name + ' - LinkMap'
+		Dialog.__init__(self, parent, parent.page.name + self.title_ending,
 			defaultwindowsize=(400, 400), buttons=Gtk.ButtonsType.CLOSE)
-		self.linkmap = linkmap
+		self.pageview = parent
```

Comment:
If NavigationModel starts returning the pageviews of new windows, is it okay to keep them in memory by holding onto them like this? It's possible to get by with just the page and the notebook without much refactoring.